lib/db/models/Node.ts

```diff
@@ -7,6 +7,16 @@ export default (sequelize: Sequelize.Sequelize, DataTypes: Sequelize.DataTypes)
     id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
     nodePubKey: { type: DataTypes.STRING, unique: true, allowNull: false },
     addressesText: { type: Sequelize.TEXT, allowNull: false },
+    lastAddressText: { type: Sequelize.TEXT, allowNull: false },
+    lastAddress: {
+      type: Sequelize.VIRTUAL,
+      get(this: db.NodeInstance) {
+        return JSON.parse(this.lastAddressText);
+      },
+      set(this: db.NodeInstance, value: Address) {
+        this.setDataValue('lastAddressText', value ? JSON.stringify(value) : '');
```

Comment:
I'd have the null case here either set an empty object `{}` or `undefined`, for the latter I think we'd need `lastAddressText` to `allowNull: true`.  Undefined would be better I think, I would expect nodes we've never had an outgoing connection to to not have a value and I suspect empty string might give us parse errors.

Reply:
so as I asked in the initiation part, I guess I need to make `lastAddress` field optional ?